packages/jsii-pacmak/lib/targets/python.ts

```diff
@@ -1182,19 +1183,15 @@ class Package {
 
     // Compute our list of dependencies
     const dependencies: string[] = [];
-    const expectedDeps = this.metadata.dependencies ?? {};
-    for (const depName of Object.keys(expectedDeps)) {
-      const depInfo = expectedDeps[depName];
-      // We need to figure out what our version range is.
-      // Basically, if it starts with Zero we want to restrict things to
-      // ~=X.Y.Z. If it does not start with zero, then we want to do ~=X.Y,>=X.Y.Z.
-      const versionParts = depInfo.version.split('.');
-      let versionSpecifier: string;
-      if (versionParts[0] === '0') {
-        versionSpecifier = `~=${versionParts.slice(0, 3).join('.')}`;
-      } else {
-        versionSpecifier = `~=${versionParts.slice(0, 2).join('.')},>=${versionParts.slice(0, 3).join('.')}`;
-      }
+    for (const [depName, version] of Object.entries(this.metadata.dependencies ?? {})) {
+      const depInfo = this.metadata.dependencyClosure![depName];
+
+      const range = new semver.Range(version);
+      const versionSpecifier = range.set.map(
+        set => set.map(
+          comp => `${comp.operator || '=='}${comp.semver.raw}`
```

Comment:
What if `comp.operator` is equal to `"="`? Shouldn't it turn to `"=="` for pip?

What if `comp.operator` is equal to `"~"`? Shouldn't it turn to `"~="`?

Reply:
Per the [type definitions](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/semver/index.d.ts#L275) `comp.operator` **cannot** be `~`. It however can be `=` in which case we'd actually render a single `=` which is no bueno. Good catch!